airbyte-api/src/main/openapi/config.yaml

```diff
@@ -1693,6 +1693,147 @@ components:
+          default: full_refresh
+        cursorField:
+          description: Path to the field that will be used to determine if a record is new or modified since the last sync. This field is REQUIRED if `sync_mode` is `incremental`. Otherwise it is ignored.
+          type: array
+          items:
+            type: string
+    AirbyteStream1:
+      type: object
+      additionalProperties: false
+      required:
+        - name
+        - json_schema
+        # todo (cgardens) - make required once sources are migrated
+        # - supported_sync_modes
+      properties:
+        name:
+          type: string
+          description: Stream's name.
+        jsonSchema:
+          description: Stream schema using Json Schema specs.
+          $ref: "#/components/schemas/StreamJsonSchema"
+        supportedSyncModes:
+          type: array
+          items:
+            $ref: "#/components/schemas/SyncMode"
+        sourceDefinedCursor:
+          description: If the source defines the cursor field, then it does any other cursor field inputs will be ignored. If it does not either the user_provided one is used or as a backup the default one is used.
+          type: boolean
+        defaultCursorField:
+          description: Path to the field that will be used to determine if a record is new or modified since the last sync. If not provided by the source, the end user will have to specify the comparable themselves.
+          type: array
+          items:
+            type: string
+    StreamJsonSchema:
+      type: object
+    # OPTION 1 END
+    # OPTION 2 START
+    AirbyteCatalog2:
+      description: describes the available schema.
+      type: object
+      required:
+        - streams
+      properties:
+        streams:
+          type: array
+          items:
+            $ref: "#/components/schemas/AirbyteStreamAndConfiguration2"
+    AirbyteStreamAndConfiguration2:
+      type: object
+      additionalProperties: false
+      properties:
+        stream:
+          $ref: "#/components/schemas/AirbyteStream2"
+        configuration:
+          $ref: "#/components/schemas/AirbyteStreamConfiguration2"
+    AirbyteStreamConfiguration2:
+      type: object
+      additionalProperties: false
+      properties:
+        syncMode:
+          $ref: "#/components/schemas/SyncMode"
+          default: full_refresh
+        cursorField:
+          description: Path to the field that will be used to determine if a record is new or modified since the last sync. This field is REQUIRED if `sync_mode` is `incremental`. Otherwise it is ignored.
+          type: array
+          items:
+            type: string
+    AirbyteStream2:
+      type: object
+      additionalProperties: false
+      required:
+        - name
+        - json_schema
+        # todo (cgardens) - make required once sources are migrated
+        # - supported_sync_modes
+      properties:
+        name:
+          type: string
+          description: Stream's name.
+        fields:
```

Comment:
note: the schema is now a tree-like structure. it is _not_ raw jsonschema, but fields must be traversed like a tree to find their children.